lib/xgu/xgux.h

```diff
@@ -0,0 +1,169 @@
+#ifndef __XGUX_H
+#define __XGUX_H
+
+#include "xgu.h"
+
+#define MIN(a,b) ((a)<(b)?(a):(b))
+#define MAX_BATCH 120
+
+inline
+void xgux_draw_arrays(XguPrimitiveType mode, unsigned int start, unsigned int count) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+    p = xgu_draw_arrays(p, start, count);
+    p = xgu_end(p);
+    pb_end(p);
+}
+
+inline
+void xgux_draw_elements16(XguPrimitiveType mode, const uint16_t* elements, unsigned int count) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+
+    /* Submit elements in pairs if possible */
+    unsigned int pair_count = count / 2;
+    for (unsigned int i = 0; i < pair_count; ) {
```

Comment:
This loop is ugly. I prefer:

```c
while(pair_count > 0) {
  [...]
  pair_count -= batch_pair_count;
  if (pair_count > 0) {
    /* Start next pushbuffer for the next iteration */
    //FIXME: This could also just be something called `pb_split();` or something
  }
}
```

Also makes it easier to translate to ASM.